internal/splunk/httprequest.go

```diff
@@ -29,6 +32,23 @@ func HandleHTTPCode(resp *http.Response) error {
 		resp.StatusCode,
 		http.StatusText(resp.StatusCode))
 
+	// Check if there is any error text returned by Splunk that we can append to the error message
+	if resp.Body != nil {
+		var jsonResponse map[string]any
+		bodyString, _ := io.ReadAll(resp.Body)
+		resp.Body = io.NopCloser(bytes.NewBuffer(bodyString))
```

Comment:
Why modify resp here?

Reply:
ReadAll drains the response body, this is a method of reinserting it back so it can be used later

Reply:
but you have the body as a string in bodyString, why would you set back the body again?

Reply:
Unless I'm missing something if we don't do this we will end up with empty `resp.Body` and it may be used higher up in some other function?

Reply:
is it though?

Reply:
Yeah, only in one place for now, but it is:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a03a2d320ead76b635f5d1f88c33d5e87a68c190/exporter/splunkhecexporter/hec_worker.go#L74

Reply:
What response codes come with the json? If 429 or 502 don't include them (I assume they don't), we don't need to execute this path and read the body

Reply:
Based on this: 

https://docs.splunk.com/Documentation/Splunk/9.2.0/Data/TroubleshootHTTPEventCollector#Possible_error_codes

It's 40*, 500 and 503. Do you mean that we should skip this whole block in case the code is different?